taichi/backends/metal/codegen_metal.cpp

```diff
@@ -1015,32 +1013,29 @@ class KernelCodegen : public IRVisitor {
     {
       ScopedIndent s2(current_appender());
       emit("const int parent_idx_ = (ii / child_num_slots);");
-      emit("if (parent_idx_ >= parent_list.num_active()) return;");
+      emit("if (parent_idx_ >= parent_list.num_active()) break;");
```

Comment:
Off the topic but critical fix: this is in the top-level Metal kernel. If we use `return`, it could skip the TLS epilogue...

Reply:
Sounds good... This reminds me of https://github.com/taichi-dev/taichi/issues/1407 - let's maybe talk about this on our next meeting :-)